composer.json

```diff
@@ -20,14 +20,16 @@
   "type": "typo3-cms-extension",
   "version": "2.2.0",
   "require": {
-    "typo3/cms": ">=6.2.0 <7.9.99"
+    "typo3/cms": "^7.6",
+    "solarium/solarium" : "^4.0"
   },
   "replace": {
     "dlf": "self.version",
     "typo3-ter/dlf": "self.version"
   },
   "autoload": {
     "classmap": [
+      "vendor/",
```

Comment:
This actually breaks the composer mode :-(
The vendor directory in typo3conf/ext/dlf/vendor will be only present in non-composer mode. In composer mode, the required libs will be installed in the project wide vendor-directory.

Reply:
OK, since it seems to be impossible to be compatible to composer mode and non-composer mode at the same time, I suggest we support composer mode as default. TYPO3 uses composer mode as preferred way of installation as well.

So, for short: Adjust composer.json and tx_emconf.php as needed for composer mode and add some documentation for people who are running in non-composer mode.

Reply:
According to [TYPO3 Core API documentation](https://docs.typo3.org/typo3cms/CoreApiReference/8.7/ApiOverview/Autoloading/) since TYPO3 7 `composer.json` is only evaluated in composer mode. In non-composer mode TYPO3 scans the whole extension directory for class files and builds its own classmap for autoloading. So we should be safe when we provide a `composer.json` for composer mode (i.e. without `vendor` directory).

Reply:
The extension's documentation should mention, that in non-composer mode you have to do a `composer update` first (to load all dependencies to `vendor/`) and (re-)install the extension via extension manager afterwards. That way all class files will be present to be scanned by the TYPO3 autoloader.

Reply:
I'm a little bit confused now... after a "composer update" we have a lot of additional thinks within the extension/vendor directory including the already mentioned typo3_src. Isn't that a problem? My suggestion is to create a valid composer.json without vendor/ and "exclude-from-classmap": ["vendor/typo3/"]". The non-composer mode for me is just dont use the command "composer" in any case. The documentation then mentions that you have to download solarium and event-dispatcher manually and place it within a subdirctory (e.G. vendor) of the extension and re-install it.

Reply:
+1 for the valid composer.json.
-1 for manual hacks ;-)

For non-composer mode, we could ship the sources of solarium and what is necessary in vendor/ OR we explain pleople to execute the `composer update` command in `typo3conf/ext/dlf/`. In the last case, also `vendor/typo3` will be downloaded but this doesn't hurt as long as it is excluded from autoloading. To exclude this, we can explicit mention the classes, paths in `ext_emconf.php` which is only used in non-composer mode.

Reply:
I just realised that there is a different autoloading behavior of T3 7.6 https://docs.typo3.org/typo3cms/CoreApiReference/7.6/ApiOverview/Autoloading/Index.html. If the extension have a composer.json it uses the autoload and psr-4 entries otherwise it scans the extension directory for classes. That was the reasen why I added a psr-4 autoloading configuration to composer.json.